libvips/foreign/cgifsave.c

```diff
@@ -203,6 +204,39 @@ vips_foreign_save_cgif_check_alpha_constraint( const VipsPel *cur,
 	return FALSE;
 }
 
+static double
+vips__cgif_compare_palettes(const VipsQuantisePalette *new,
+	const VipsQuantisePalette *old)
+{
+	g_assert( new->count <= 256 );
+	g_assert( old->count <= 256 );
+
+	int i, j;
+	double total_dist = 0.0;
+
+	for( i = 0; i < new->count; i++ ) {
+		double best_dist = 255.0 * 255.0 * 3;
+
+		for( j = 0; j < old->count; j++ ) {
+			if(
+				(new->entries[i].a >= 128) !=
+				(old->entries[j].a >= 128)
+			) continue;
```

Comment:
I think this will ignore palette entries which stay transparent or which stay solid. You'll only consider entries which have changed from solid to transparent, or transparent to solid.

Don't you want to only consider pixels which are solid now and were also solid in the previous frame?